airflow/kubernetes/pod_generator.py

```diff
@@ -370,6 +368,10 @@ def construct_pod(  # pylint: disable=too-many-arguments
         except Exception:  # pylint: disable=W0703
             image = kube_image
 
+        task_id = make_safe_label_value(task_id)
+        dag_id = make_safe_label_value(dag_id)
+        scheduler_job_id = make_safe_label_value(scheduler_job_id)
```

Comment:
@houqp `scheduler_job_id` needs to be converted to a string.

Reply:
scheduler_job_id is annotated as `scheduler_job_id: str,` in construct_pod's function signature as well as in `AirflowKubernetesScheduler` and many other class/function type signatures, do we need to correct the type annotation for all of them?

Reply:
I'll try to look into this a bit more today and see if I can get a better handle on why the type here is not the expected `str`.

Reply:
i do think it would be more consistent if we use int everywhere since that's the type we get from db.

Reply:
I have added the `str` conversion, perhaps we should address the type annotation fix in a separate followup PR